uds/diagnostic_configuration/ecu_configuration.py

```
@@ -0,0 +1,334 @@
+        except (ValueError, KeyError):
+            dids = set()
+            rids = set()
+        else:
+            dids = self.__extract_dids(decoded_message=decoded_message)
+            rids = self.__extract_rids(decoded_message=decoded_message)
+        subfunction = self.__extract_subfunction(message_payload=message_payload)
+        restrictions = [self.sid_restrictions[sid]]
+        if subfunction is not None:
+            restrictions.append(self.subfunction_restrictions[sid][subfunction])
+        for did in dids:
+            restrictions.append(self.did_restrictions[sid][did])
```

Ruff (0.15.20)

[warning] Use `list.extend` to create a transformed list

(PERF401)